File: Problem/codility/codility_Lesson02_Q2.cpp
#include <iostream>
#include <algorithm>
#include <cstdio>
#include <vector>
#include <string>

using namespace std;


vector<int> solution(vector<int> &A, int K) {
vector<int> res(A.size());

for (int i = 0; i < A.size(); i++) {
res[(i + K) % A.size()] = A[i];
}

return res;
}

int main() {
vector<int> A = { 3,8,9,7,6 };

vector<int> res = solution(A, 3);
}

File: Problem/codility/codility_Lesson02_Q2.java
import java.util.*;

class Solution {
    public int solution(int[] A) {
        HashMap<Integer,Integer> map = new HashMap<Integer,Integer>();

        for(int i=0;i<A.length;i++){
            if(map.get(A[i]) == null){
                map.put(A[i],1);
            }else{
                map.put(A[i],map.get(A[i])+1);
            }
        }

        int res = 0;
        for(int key : map.keySet()){
            if(map.get(key) % 2 != 0){
                res = key;
            }
        }
        return res;
    }
}